app/controllers/errors_controller.rb

```diff
@@ -47,29 +37,55 @@ def notify_subscribers
   end
 
   def resolve
-    resolve_issues(params[:error_ids], true, params[:per_page], params[:page], @error.id)
-  end
-
-  def unresolve
-    resolve_issues(params[:error_ids], false, params[:per_page], params[:page], @error.id)
+    resolve_issues(params[:error_ids])
   end
 
   private
 
-  def resolve_issues (ids, resolve, errors_per_page, page, current_error)
-    errors_per_page ||= 5
-    page ||= 1
-    errors = GroupedIssue.find(current_error).website.grouped_issues.order('last_seen DESC')
-    if resolve
+  def resolve_issues(ids)
+    #flatten array before passing it to activerecord since its gonna be deprecated
+    ids.flatten!
+    errors_per_page = params[:errors_per_page] || 5
+    page = params[:page] || 1
+    errors = current_website.grouped_issues.order('last_seen DESC')
+    if resolved?
       GroupedIssue.where(id: ids).update_all(resolved_at: Time.now.utc, status: GroupedIssue::RESOLVED)
-      errors = errors.with_status(:unresolved)
+      errors = errors.with_status(:resolved)
     else
       GroupedIssue.where(id: ids).update_all(resolved_at: nil, status: GroupedIssue::UNRESOLVED)
-      errors = errors.with_status(:resolved)
+      errors = errors.with_status(:unresolved)
+    end
+
+    if ids.include?(@error.id.to_s)
+      @error = @error.reload
+      position = errors.where("last_seen >= ?", @error.last_seen).count
+      page = (position.to_f/errors_per_page).ceil
+      redirect_to error_path(@error, "#{ @error.resolved? ? 'resolved' : 'unresolved' }": true, page: page)
+    else
+      redirect_to error_path(@error, "#{ resolved? ? 'unresolved' : 'resolved' }": true)
+    end
+  end
+
+  def current_issue_page
+    page = params[:page] || 1
+    errors_per_page = 5
+    errors = current_website.grouped_issues.order('last_seen DESC')
+    if resolved?.nil? && params[:page].nil?
```

Comment:
here i would prefer if you would do `params[:tab].present?` instead of doing `resolved?.nil?` that is so weird in the way it's written.